### PR TITLE
[5.2] Add languages API endpoint

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
fix  for GET `{{base_path}}/api/index.php/v1/languages`

### Testing Instructions
 GET `{{base_path}}/api/index.php/v1/languages`

### Actual result BEFORE applying this Pull Request
500 Internal server error

### Expected result AFTER applying this Pull Request
200 OK

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on d87a793249d1b8a94e8677b34bb90adf2a05f8d8<br /><br />Tested successfully in Joomla 4.4.0-rc2-dev of 14 October using PHP 8.2.11 in Wampserver 3.3.1 and Postman Desktop Agent<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42136">issues.joomla.org/tracker/joomla-cms/42136</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d87a793249d1b8a94e8677b34bb90adf2a05f8d8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42136">issues.joomla.org/tracker/joomla-cms/42136</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42136">issues.joomla.org/tracker/joomla-cms/42136</a>.</sub>

File: api/components/com_installer/src/View/Languages/JsonapiView.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     * @return  object
+     *
+     * @since   4.0.0
+     */
+    protected function prepareItem($item)
+    {
+        $item->id = ++$this->i;
+        return parent::prepareItem($item);
+
+    }
```
Comment (Quy) [inline api/components/com_installer/src/View/Languages/JsonapiView.php]:
```suggestion
        $item->id = ++$this->i;

        return parent::prepareItem($item);
    }
```

File: api/components/com_installer/src/Controller/LanguagesController.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use Tobscure\JsonApi\Exception\InvalidParameterException;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * The manage controller
+ *
+ * @since  4.0.0
```
Comment (Quy) [inline api/components/com_installer/src/Controller/LanguagesController.php]: `__DEPLOY_VERSION__`

**Comment by laoneo** [comment]
Please wait till 5.1 is fully up to date with 5.0.

**Comment by alikon** [comment]
this pr is not a **feature** it'a a bug fix
i'm not adding any new endpoint here i'm just fixing response HTTP 500
see https://docs.joomla.org/J4.x:Joomla_Core_APIs

**Comment by laoneo** [comment]
For me it looked like a new feature. Next time it would help when you update the pr description with the link.

File: api/components/com_installer/src/View/Languages/JsonapiView.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @since  4.0.0
+ */
+class JsonapiView extends BaseApiView
+{
+    /**
+     * The fields to render item in the documents
+     *
+     * @var  array
+     * @since  4.0.0
```
Comment (Quy) [inline api/components/com_installer/src/View/Languages/JsonapiView.php]: `__DEPLOY_VERSION__`

Comment (alikon) [reply api/components/com_installer/src/View/Languages/JsonapiView.php]: oops

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7390b4ed23003b743392cece00e0426710a5dc1d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42136">issues.joomla.org/tracker/joomla-cms/42136</a>.</sub>

**Comment by nielsnuebel** [comment]
I have tested this item :white_check_mark: successfully on 7390b4ed23003b743392cece00e0426710a5dc1d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42136">issues.joomla.org/tracker/joomla-cms/42136</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42136">issues.joomla.org/tracker/joomla-cms/42136</a>.</sub>

**Comment by alikon** [comment]
no interest

**Comment by pe7er** [comment]
Thanks @alikon !

**Pull request merged**